src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String SEPARATOR = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalSupply = calculate(fromFileName, SUPPLY);
+        int totalBuy = calculate(fromFileName, BUY);
+        int result = totalSupply - totalBuy;
+        writeAccount(toFileName, totalSupply, totalBuy, result);
+    }
+
+    private int calculate(String fileName, String operationType) {
+        int total = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                total += convertLine(line, operationType);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file " + fileName, e);
+        }
+        return total;
+    }
+
+    private int convertLine(String line, String operationType) {
```

Comment:
The method 'convertLine' name could be more descriptive, it's not immediately clear what 'convert' means in this context. Consider renaming it to reflect that it's parsing and summing values of a certain operation type.